### ISSUE TITLE
regression regex (due to #740)

### ISSUE THREAD

**Issue by xuwei-k** [scala-native/scala-native#886]
## JVM

```
Welcome to Scala 2.11.11 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_131).
Type in expressions for evaluation. Or try :help.

scala> val JsonNumberRegex = """(-)?((?:[1-9][0-9]*|0))(?:\.([0-9]+))?(?:[eE]([-+]?[0-9]+))?""".r
JsonNumberRegex: scala.util.matching.Regex = (-)?((?:[1-9][0-9]*|0))(?:\.([0-9]+))?(?:[eE]([-+]?[0-9]+))?

scala> val JsonNumberRegex(a,b,c,d) = "0.00000"
a: String = null
b: String = 0
c: String = 00000
d: String = null
```

## scala-native 0.3.0

```scala
object Main {
  def main(args: Array[String]): Unit = {
    val JsonNumberRegex = """(-)?((?:[1-9][0-9]*|0))(?:\.([0-9]+))?(?:[eE]([-+]?[0-9]+))?""".r
    val JsonNumberRegex(negative, intStr, decStr, expStr) = "0.000000"
    assert(negative == null) // success or StringIndexOutOfBoundsException
  }
}
```

## scala-native 0.3.1

```scala
object Main {
  def main(args: Array[String]): Unit = {
    val JsonNumberRegex = """(-)?((?:[1-9][0-9]*|0))(?:\.([0-9]+))?(?:[eE]([-+]?[0-9]+))?""".r
    val JsonNumberRegex(negative, intStr, decStr, expStr) = "0.000000"
    assert(negative == null) // fail. "negative" is empty string
  }
}
```

ref #740 , related #873 ?
/cc @muxanick

**Comment by muxanick** [comment]
I will create PR for fix tomorrow

**Comment by muxanick** [comment]
Thanks for the bug report!

**Comment by muxanick** [comment]
PR #891

**Comment by densh** [comment]
We'll ship the fix for this one in 0.3.2.
